sematic/db/models/json_encodable_mixin.py

```diff
@@ -16,41 +16,37 @@ class JSONEncodableMixin:
     Defines JSON-encodable behavior.
     """
 
-    def to_json_encodable(self):
+    def to_json_encodable(self, redact: bool = True):
         return {
             column.key: _to_json_encodable(getattr(self, column.key), column)
-            for column in self.__table__.columns
-            if not column.info.get(REDACTED_KEY, False)
+            for column in self.__table__.columns  # type: ignore
+            if (not column.info.get(REDACTED_KEY, False) or not redact)
         }
 
     @classmethod
     def from_json_encodable(cls, json_encodable):
         field_dict = {
             column.key: cls.field_from_json_encodable(column.key, json_encodable)
             for column in inspect(cls).attrs
+            if column.key in json_encodable
```

Comment:
This is the only reason resolution decoding works with the new server, and it executes on clients. Therefore the change in redaction behavior is backwards incompatible with old clients. Please either:

(a) leave resolution redaction as it was, but roll out this client. Then in a few versions we can change resolution redaction to the new mechanism so we have a window of compatibility OR
(b) Bump `MIN_CLIENT_SERVER_SUPPORTS` in `versions.py` so the next release has a window of compatibility of 0 versions

I prefer (a) since we have users who have requested that we try to keep at least a 1 version-wide window of compatibility

Reply:
What is the reason for completely omitting the key if the field is redacted? Esthetics? Can we keep the old behavior to avoid this issue?

Reply:
I would rather users not even know that we have an `api_key` field on the `User` model. Same for resolutions env vars.
Additionally, if we want to leave the keys, we have to tap into SQLAlchemy default mechanism to populate the field with a type-compatible default.
Removing the field altogether is cleaner, safer, and easier.